### PR TITLE
Fix TypeError « Cannot read property 'y' of undefined

### PR DESCRIPTION
**Pull request by jcharlaix**

### CODE REVIEW

File: src/Viewer/Panels/Graph/graph.js [hunk 1]
```diff
@@ -305,6 +305,7 @@ const Chart = sceneWrapper(({ valueTransformFn, ifManualData, data, unit, contai
     data.automated.length,
     (data.manual ? data.manual.length : 0)
   )
+  const nb = vertiaclTickNb - 1
```
Comment (naittaleb) [inline src/Viewer/Panels/Graph/graph.js]: Could you expalin by a brief comment why do you substract 1 to nb?

File: src/Viewer/Panels/index.js [hunk 2]
```diff
@@ -59,18 +59,17 @@ export default function Panels () {
   const panelsData = useMemo(() => {
     const tempFruitPoints = get(scan, 'data.angles.fruit_points')
     const fruitPoints = tempFruitPoints
-      ? tempFruitPoints.slice(0, tempFruitPoints.length - 1)
-      : undefined
-
+    // ? tempFruitPoints.slice(0, tempFruitPoints.length - 1)
```
Comment (naittaleb) [inline src/Viewer/Panels/index.js:62]: If these code lines are useless. You can just remove them

File: src/Viewer/Panels/index.js [hunk 3]
```diff
@@ -59,18 +59,17 @@ export default function Panels () {
...
+    // : undefined
     const tempAutomatedAngles = get(scan, 'data.angles.angles')
     const automatedAngles = tempAutomatedAngles
-      ? tempAutomatedAngles.slice(0, tempAutomatedAngles.length - 1)
-      : undefined
+    // ? tempAutomatedAngles.slice(0, tempAutomatedAngles.length - 1)
```
Comment (naittaleb) [inline src/Viewer/Panels/index.js:66]: same comment

File: src/Viewer/Panels/index.js [hunk 4]
```diff
@@ -59,18 +59,17 @@ export default function Panels () {
...
+    // : undefined
 
     const tempAutomatedInternodes = get(scan, 'data.angles.internodes')
     const automatedInternodes = tempAutomatedInternodes
-      ? tempAutomatedInternodes.slice(0, tempAutomatedInternodes.length - 1)
-      : undefined
+    // ? tempAutomatedInternodes.slice(0, tempAutomatedInternodes.length - 1)
```
Comment (naittaleb) [inline src/Viewer/Panels/index.js:71]: same

**Review by fabfabBesnard — APPROVED** [review]
PR validated !

The changes fix the issue on at least one dataset analysis:
-dataset 2018-12-10_14-12-59 (Ler0_19_10_2018_A)
-analysis with plant-3d-vision v0.9 / geom pipe / config file = geom_pipe_real.toml modified with stem_axis_inverted = true in task [TreeGraph] + [AnglesAndInternodes]
-Ground truth (manual) measure has 30 intervals (31 fruits); analysis finds 31 interval / 32 fruits (Julie identified that the error typically occurs when nber of computed measure > nb manual measures)
-Plant-3d-explorer -> with current master, error "Cannot read property 'y' of undefined"
                             -> with this branch: everything works well !

image

**Review by naittaleb — CHANGES_REQUESTED** [review]

**Comment by fabfabBesnard** [comment]
ben du coup, on la merge celle-là ? C'est bon pour vous Julie et Nabil ?

**Review by naittaleb — APPROVED** [review]

**Pull request merged**